src/main/resources/assets/computercraft/lua/rom/apis/parallel.lua

```diff
@@ -31,36 +49,82 @@ local function runUntilLimit( _routines, _limit )
                     else
                         tFilters[r] = param
                     end
-                    if coroutine.status( r ) == "dead" then
-                        _routines[n] = nil
-                        living = living - 1
-                        if living <= _limit then
-                            return n
+                    if _data then
+                        local last_count = count
+                        run, count = _data()
+                        if _limit > 0 then
+                            _limit = _limit+(count-last_count)
                         end
                     end
+                    local c = checkDead(r, n)
+                    if c then return c end
                 end
             end
         end
-        for n=1,count do
-            local r = _routines[n]
-            if r and coroutine.status( r ) == "dead" then
-                _routines[n] = nil
-                living = living - 1
-                if living <= _limit then
-                    return n
-                end
-            end
+        for n = 1, count do
+            local c = checkDead(_routines[n], n)
```

Comment:
It's probably also worth dropping dead coroutines from the coroutine list.

Reply:
Or use the null object pattern? Have a fake coroutine which doesn't go dead it just yields in a loop. If all of the coroutines in the list are this fake one then all of the original ones are dead. Saves having to resize the list.